app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -48,8 +50,8 @@ def accessible_puppet_proxies(obj)
     list
   end
 
-  def domain_subnets
-    accessible_related_resource(@domain, :subnets)
+  def domain_subnets(type)
+    accessible_related_resource(@domain, :subnets, :query => {:type => type})
```

Comment:
@tbrisker: I couldn't do it because `return [] if obj.blank?` would lead to `[].where(:type => type)` and that obviously does not work. Is there a better approach?